File: JavaToKotlin/app/src/main/java/io/dustin/apps/common/utils/OptionalUtils.java
package io.dustin.apps.common.utils;

import io.dustin.apps.common.exception.DataNotFoundException;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

import java.util.Optional;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class OptionalUtils {


    public static <T> T getEntity(Optional<T> optional, Class<T> clazz, String errorMessage) {
        if(optional.isPresent()) {
            return optional.get();
        } else {
            throw new DataNotFoundException("데이터가 없습니다.");
        }
    }

}

File: JavaToKotlin/apps(kotlin)/src/main/kotlin/io/dustin/apps/common/utils/OptionalUtils.kt
package io.dustin.apps.common.utils

import io.dustin.apps.common.exception.DataNotFoundException

@NoArgsConstructor(access = AccessLevel.PRIVATE)
object OptionalUtils {
    fun <T> getEntity(optional: Optional<T?>?, clazz: Class<T?>?, errorMessage: String?): T? {
        return if (optional.isPresent()) {
            optional.get()
        } else {
            throw DataNotFoundException("데이터가 없습니다.")
        }
    }
}
